### ISSUE TITLE
Suggestion: Move Header-writing to initialization as much as possible

### ISSUE THREAD

**Issue by tminich** [itsgoingd/clockwork#384]
Currently headers are written in requestProcessed, which is called at the very end of dealing with a request. In an ideal world this is no issue, but sometimes you have to deal with dirty legacy code and you run into the dreaded 'Headers already sent' error and have no easy way to fix it.

Since in most cases all clockwork info necessary to be written into the headers seems to be available at initialization time, why not write those headers then? 
As far as I can see from the source, the only header that needs collected data is the Server-Timing header, and that is written only if set in the config, so it doesn't cause an issue if you don't request it.

Since I regularly deal with legacy code moving writing of the essential headers to initialization time would be highly appreciated :)

**Comment by itsgoingd** [comment]
Hey, I see what you mean.

I think we could extract the headers sending code to a separate method and make it optional in `requestProcessed`, eg.:

```php
$clockwork = Clockwork::init();
$clockwork->appendClockworkHeaders();
...
$clockwork->requestProcessed(false);
```

I'll try to take a look at this soon.

**Comment by tminich** [comment]
Awesome. Since the headers need to be sent to make clockwork work I'd suggest tracking it with a class or instance variable (eg like headersSent) so they can be sent in requestProcessed if the user hasn't already done so manually.

**Comment by itsgoingd** [comment]
Good idea. I'd also accept a PR if you want to make one.

**Comment by itsgoingd** [comment]
Now available in Clockwork 4.1.
